CvGameCoreDLL_Expansion2/CvCity.cpp

```diff
@@ -6603,7 +6603,7 @@ CvString CvCity::GetDisabledTooltip(CityEventChoiceTypes eChosenEventChoice, int
 }
 void CvCity::DoEventChoice(CityEventChoiceTypes eEventChoice, CityEventTypes eCityEvent, bool bSendMsg, int iEspionageValue, PlayerTypes eSpyOwner)
 {
-	if (GC.getGame().isNetworkMultiPlayer() && bSendMsg && GET_PLAYER(getOwner()).isHuman()) {
+	if (GC.getGame().isNetworkMultiPlayer() && bSendMsg && (GET_PLAYER(getOwner()).isHuman() || (eSpyOwner != NO_PLAYER && GET_PLAYER(eSpyOwner).isHuman()))) {
```

Comment:
Also added this missing check. Also not tied with the issue. Looks like when you are sending spy to some AI city such events are not broadcasted at all because initial check `GET_PLAYER(getOwner()).isHuman()` is false.